pkg/virt-handler/vm.go

```diff
@@ -2689,6 +2693,26 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 	return nil
 }
 
+func (d *VirtualMachineController) hotplugSriovInterfaces(vmi *v1.VirtualMachineInstance) error {
+	sriovInterfaces := netvmispec.FilterSRIOVInterfaces(vmi.Spec.Domain.Devices.Interfaces)
+	sriovStatusInterfaces := netvmispec.FilterStatusInterfacesByNameSet(vmi.Status.Interfaces, netvmispec.GetInterfacesNames(sriovInterfaces))
+	if len(sriovInterfaces) == len(sriovStatusInterfaces) {
+		return nil
+	}
+
+	client, err := d.getVerifiedLauncherClient(vmi)
+	if err != nil {
+		return err
```

Comment:
If you exit from this function with an error, please add text that mention that hotplug SR-IOV interfaces failed.

Reply:
On the caller, the err is logged with a proper message

Reply:
Well, I would expect the information about why it failed to be provided at this level, so the upper level will express only data it knows and nothing else. If you add it here, the caller (or callers) will not need to add any extra data.

Reply:
Did you mean that the function should return wrapped error with `failed to hot-plug sriov interfaces` instead of wrapping the error on the caller?
Note that currently log message will look like `failed to hot-plug sriov interfaces: <returned error>`.
To me it make more sense to wrap the returned error on the caller instead of wrapping each returned error inside the function, WDYT?

Reply:
If each abstraction layer handles its own information collection about the error, then we do not need to do the same from each caller. A method may be called by multiple callers, having each writing the error message, potentially in an inconsistent manner, is not that good and causes duplication for no reason.
I also do not know why it makes sense to you to see a caller calling `foo` and then adding the information about `foo` failing. If `foo` failed, it makes more sense to me that it will provide all possible information about it, including its own action as failed.

The format of how the log looks is not suppose to change if you do not add anything.

Reply:
Alright, thanks for explaining! I will defiantly take this in account for next PR's.

Reply:
Done